lib/Crypto/Encryption.php

```diff
@@ -552,6 +552,9 @@ protected function getPathToRealFile($path) {
 			$realPath = '/' . $parts[1] . '/files/' . \implode('/', \array_slice($parts, 3));
 			$length = \strrpos($realPath, '.');
 			$realPath = \substr($realPath, 0, $length);
+		} elseif ($parts[2] === 'files_trashbin' && $parts[3] === 'versions') {
+			$realPath = "/{$parts[1]}/files_trashbin/files/" . \implode('/', \array_slice($parts, 4));
+			$realPath = \preg_replace('/.v[0-9]+([^\/]*)$/', '$1', $realPath);
 		}
 
 		return $realPath;
```

Comment:
@jvillafanez maybe I am too picky, but would not it be better if the if-else block actually checked all possible combinations and final else throws an exception on unsupported? This way we could avoid an issues like this one. Thoughts?

Reply:
This piece of code also runs for regular files, and for those files the expected path matches the original, so there is nothing to do. Even if we throw an exception here, we'll need to handle the key location for the regular files. I think it's better for the code to go through this function in order to handle all the cases in the same place.